src/main/java/liuni/dagger/ServiceInjectionModule.java

```diff
@@ -0,0 +1,30 @@
+package liuni.dagger;
+
+import dagger.Module;
+import dagger.Provides;
+import liuni.configs.LiUniConfig;
+import liuni.services.TwitterService;
+
+import javax.inject.Singleton;
+
+@Module
+public class ServiceInjectionModule {
+    private static TwitterService twitterService;
+    LiUniConfig config;
+
+    public ServiceInjectionModule(LiUniConfig config) {
+        this.config = config;
+    }
+
+    @Provides
+    @Singleton
+    TwitterService provideTwitterService() {
+        if (twitterService == null) {
```

Comment:
you shouldn't need the static variable as this is all taken care of by dagger

Reply:
Since it is a Singleton, don't we want to only create it if it is null?

Reply:
Done